packages/subiquity_client/lib/subiquity_server.dart

```diff
@@ -64,7 +66,15 @@ abstract class SubiquityServer {
       ...?args,
     ];
 
-    var subiquityPath = p.join(Directory.current.path, 'subiquity');
+    // try using local subiquity
+    String? subiquityPath = p.join(Directory.current.path, 'subiquity');
+    if (subiquityPath != null) {
+      bool exists = await Directory(subiquityPath).exists();
```

Comment:
there's also `existSync()` which is rumoured to be 10x faster because it doesn't need to go through the event loop. a blocking call would be fine here because the UI is not running yet.

Reply:
oh, and since p.join() returns a non-nullable value, we could leave out the null check and use the bang operator: `Directory(subiquityPath!)`

Reply:
> oh, and since p.join() returns a non-nullable value, we could leave out the null check and use the bang operator: Directory(subiquityPath!)

@jpnurmi: this was exactly my first try, which lead to this linter error: `warning • The '!' will have no effect because the receiver can't be null • lib/subiquity_server.dart:71:48 • unnecessary_non_null_assertion`
See https://github.com/canonical/ubuntu-desktop-installer/runs/3689760243. Line 71 in this commit was:
`bool exists = await Directory(subiquityPath!).exists();`

Reply:
it looks like the analyzer isn't fully satisfied
```
   info • The operand can't be null, so the condition is always true • lib/subiquity_server.dart:71:23 • unnecessary_null_comparison
   info • Omit type annotations for local variables • lib/subiquity_server.dart:72:7 • omit_local_variable_types
```